### PR TITLE
Refs task #390978 #387270 Unit test Search & list users

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets

- [#390978](https://dev.sun-asterisk.com/issues/390978)
- [#387270](https://dev.sun-asterisk.com/issues/387270)

## What's this PR do ?

- [x] Write unit test for list users
- [x] Write unit test for search users

## Library
*(List package, library third party add new include version)*

- N/A

## Performance

- [ ] Resolved n + 1 query
- [ ] Run explain query already
- [ ] Time run rake task : 1000 ms

## Checklist

- [x] It was tested in local success?
- [x] Fill link PR into ticket and the opposite
- [x] Note purpose/reason, solution, scope of influence into ticket

## ENV note

```
```

## Notes
*(Other notes)*
image

### CODE REVIEW

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr></table>

File: tests/Feature/UserControllerTest.php [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+       $response->assertStatus(200);
+       $response->assertViewHas('users');
+   }
+
+   /**
+    * test Feature list user when user not login
+    *
+    * @return void
+    */
+   public function testListUserUnauthorizedFeature()
```
Comment (nguyenthinhhn) [inline tests/Feature/UserControllerTest.php]: chỗ này là check Unauthenticate chứ nhỉ a

**Review by quanghung97 — APPROVED** [review]
OK

**Review by nguyenthinhhn — APPROVED** [review]

File: tests/Unit/Repositories/UserRepositoryTest.php [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+     */
+    public function testGetAllAndSearchWithoutKeyword()
+    {
+        factory(User::class, 2)->create();
+
+        $userRepository = new UserRepository;
+        $perPage = config('paginate.perPage');
+
+        $this->assertCount(2, $userRepository->getAllAndSearch($perPage, null));
+
```
Comment (ghost) [inline tests/Unit/Repositories/UserRepositoryTest.php:64]: remove line naf y di chu

**Review by ghost — APPROVED** [review]

**Review by tamtt-1031 — APPROVED** [review]

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr></table>

**Comment by phucqd-1305** [comment]
OK

**Pull request merged**